source/Mlos.Python/mlos/Optimizers/BayesianOptimizerFactory.py

```diff
@@ -24,12 +25,28 @@ class BayesianOptimizerFactory:
         Logger object
     """
 
-    def __init__(self, grpc_channel, logger=None):
+    def __init__(self, grpc_channel=None, logger=None):
         self.logger = logger if logger is not None else create_logger("BayesianOptimizerFactory")
         self._grpc_channel = grpc_channel
-        self._optimizer_service_stub = OptimizerServiceStub(channel=self._grpc_channel)
+        self._optimizer_service_stub = None
+        if self._grpc_channel is not None:
+            self._optimizer_service_stub = OptimizerServiceStub(channel=self._grpc_channel)
 
-    def create_remote_optimizer(self, optimization_problem: OptimizationProblem, optimizer_config: Point = None) -> BayesianOptimizerProxy:
+
+    def create_local_optimizer(self, optimization_problem: OptimizationProblem, optimizer_config: Point = None) -> OptimizerBase:

```

Comment:
Should the type returned by the BayesianOptimizerFactory be a BayesianOptimizer, or does that mess with the class hierarchy too much.

Reply:
Yeah, that will work.